library/lib.sh

```diff
@@ -257,35 +255,69 @@ lsrConvertToCollection() {
 
 lsrGetManagedNodes() {
     local guests_yml=$1
-    yq -r ". | keys[] | select(test(\"managed*\"))" "$guests_yml" | sort
+    sed --quiet --regexp-extended 's/(^managed.*):/\1/p' "$guests_yml" | sort
 }
 
 lsrGetNodes() {
     local guests_yml=$1
-    yq -r ". | keys[]" "$guests_yml" | sort
+    sed --quiet --regexp-extended 's/(^[^ ]*):/\1/p' "$guests_yml" | sort
+}
+
+lsrGetNodeName() {
+    local guests_yml=$1
+    local node_pat=$2
+    sed --quiet --regexp-extended "s/(^$node_pat.*):/\1/p" "$guests_yml"
+}
+
+lsrGetCurrNodeHostname() {
+    local guests_yml=$1
+    local ip_addr
+    ip_addr=$(hostname -I | awk '{print $1}')
+    grep "primary-address: $ip_addr" "$guests_yml" -B 10 | sed --quiet --regexp-extended 's/(^[^ ]*):/\1/p'
+}
+
+lsrGetNodeIp() {
+    local guests_yml=$1
+    local node=$2
+    # awk '$1=$1' to remove extra spaces
+    sed --quiet "/$node\:/,/^[^ ]/p" "$guests_yml"  | sed --quiet --regexp-extended 's/primary-address: (.*)/\1/p' | awk '$1=$1'
```

Comment:
```suggestion
    sed --quiet "/$node\:/,/^[^ ]/p" "$guests_yml"  | sed --quiet --regexp-extended 's/primary-address: (.*)/\1/p' | awk '$1=$1'
```
why the `node\:`?  if the problem is using `$node:` then you can use `${node}:` instead

Reply:
\: is not needed, colon without a backslash works just fine, removing this. Thank you